project_stock/models/project_task.py

```diff
@@ -220,13 +220,7 @@ def action_done(self):
             lambda x: x.state not in ("done", "cancel")
         ):
             move.quantity_done = move.reserved_availability
-        moves_todo = self.mapped("move_ids")._action_done()
-        # Use sudo to avoid error for users with no access to analytic
-        analytic_line_model = self.env["account.analytic.line"].sudo()
-        for move in moves_todo:
-            vals = move._prepare_analytic_line_from_task()
-            if vals:
-                analytic_line_model.create(vals)
+        self.mapped("move_ids")._action_done()
```

Comment:
No need of the `mapped` with relational fields.

```suggestion
        self.move_ids._action_done()
```